make/git/hook-pre-commit.sh

```diff
@@ -57,4 +57,30 @@ if [ -z "${existing}" ]; then
     printf "Please double-check that it is configured properly.\n"
 fi
 
+# Prevent mistaken submodule commits by checking whether committing
+# submodule + other stuff and if so bail if the submodule version is older.
+# We want to allow a submodule-only rollback.
+# XXX: this relies on several unix utilities which we assume are available
+# on Windows.
```

Comment:
Looks like a close copy from the drmemory version: but there's no really simple way to share unfortunately.  Maybe add a XXX comment that this is mostly a verbatim copy and if we set up some way to share (dedicated separate repository that's a submodule in both that provides shared config files or sthg?) we should be sure to share this too.

Reply:
Done.